From 38a7d96e01f897f1da38d1fcbcbc362fc6628052 Mon Sep 17 00:00:00 2001
From: Pavel Tikhomirov <ptikhomirov@virtuozzo.com>
Date: Wed, 8 Jul 2020 09:57:33 +0300
Subject: [PATCH 185/249] mount: fix broken remounted_rw check

Expression (x && REMOUNTED_RW) is always same as just (x).

It should've been (x & REMOUNTED_RW) to check if mount is marked as
temporary remounted writable and requires to be switched back.

By fixing this check we eliminate excess readonly remounts.

Cherry-picked from Virtuozzo criu:
https://src.openvz.org/projects/OVZ/repos/criu/commits/167f8ac67

Signed-off-by: Pavel Tikhomirov <ptikhomirov@virtuozzo.com>
---
 criu/mount.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/criu/mount.c b/criu/mount.c
index b2810b7a5..3421ef491 100644
--- a/criu/mount.c
+++ b/criu/mount.c
@@ -4113,7 +4113,7 @@ static int __remount_readonly_mounts(struct ns_id *ns)
 		if (ns && mi->nsid != ns)
 			continue;
 
-		if (!(*mi->remounted_rw && REMOUNTED_RW))
+		if (!(*mi->remounted_rw & REMOUNTED_RW))
 			continue;
 
 		/*
-- 
2.35.1

